core/common/hexutil.hpp

```diff
@@ -62,6 +66,38 @@ namespace kagome::common {
    */
   outcome::result<std::vector<uint8_t>> unhexWith0x(std::string_view hex);
 
+  /**
+   * @brief unhex hex-string with 0x or without it in the beginning
+   * @tparam T unsigned integer value type to decode
+   * @param value source hex string
+   * @return unhexed value
+   */
+  template <class T, typename = std::enable_if<std::is_unsigned_v<T>>>
+  outcome::result<T> unhexNumber(std::string_view value) {
+    std::vector<uint8_t> bytes;
+    static const std::string leading_chars = "0x";
+    if (value.substr(0, 2) == leading_chars) {
+      OUTCOME_TRY(bts, common::unhexWith0x(value));
+      bytes = std::move(bts);
+    } else {
+      OUTCOME_TRY(bts, common::unhex(value));
+      bytes = std::move(bts);
+    }
+
+    if (bytes.size() > sizeof(T)) {
+      return UnhexError::VALUE_OUT_OF_RANGE;
+    }
```

Comment:
This is incomplete check. Remember signed/unsigned numbers.
Suggest to use `intmax_t` as temporary, and compare it with `std::numeric_limit<T>::max()` and `...::min()`.

Reply:
template <class T, typename = std::enable_if<std::is_unsigned_v<T>>>